lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -1316,6 +1316,86 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     return b.create<arith::XOrIOp>(loc, payloadArgs[0], allOnesVal);
   }
 
+  if (isa<AtenDequantizeTensorOp, AtenDequantizeSelfOp>(op)) {
+    auto value = payloadArgs[0];
+    auto qtensor = op->getOperand(0);
+    auto qtensorTy = qtensor.getType().cast<ValueTensorType>().getDtype();
+    auto makeQTensor =
+        qtensor.getDefiningOp<Aten_MakePerTensorQuantizedTensorOp>();
+    if (!makeQTensor) {
+      qtensor.dump();
+      op->emitError("dequantizing tensor of unknown scale / zeorpoint");
+      return nullptr;
+    }
+
+    if (torch_to_linalg::isUnsignedTorchType(qtensorTy)) {
+      value = b.create<arith::UIToFPOp>(loc, payloadArgs[1].getType(), value);
+    } else {
+      value = b.create<arith::SIToFPOp>(loc, payloadArgs[1].getType(), value);
+    }
+
+    Value zp = makeQTensor.getZeroPoint();
+    zp = converter->materializeTargetConversion(
+        b, loc, converter->convertType(zp.getType()),
+        makeQTensor.getZeroPoint());
+    zp = b.create<arith::SIToFPOp>(loc, value.getType(), zp);
+    value = b.create<arith::SubFOp>(loc, value, zp);
```

Comment:
If we're converting both the value and the zero point to floating point, could we instead do the subtraction on integers and then do the conversion? Maybe signed/unsigned extending to the bitwidth of the target floating point type first. Just a suggestion, and not sure that it really matters.

Reply:
Performed in fixed point math. It should not technically matter except in extreme cases of `int32` dequantization, but this is good enough.

Reply:
Sounds good to me